arch/xtensa/src/esp32s3/esp32s3_start.c

```diff
@@ -248,15 +248,15 @@ void noreturn_function IRAM_ATTR __esp32s3_start(void)
 
   /* Move CPU0 exception vectors to IRAM */
 
-  __asm__ __volatile__ ("wsr %0, vecbase\n"::"r" (&_init_start));
+  __asm__ __volatile__ ("wsr %0, vecbase\n"::"r" (_init_start));
 
   /* Clear .bss. We'll do this inline (vs. calling memset) just to be
    * certain that there are no issues with the state of global variables.
    */
 
-  for (uint32_t *dest = &_sbss; dest < &_ebss; dest++)
+  for (uint32_t dest = (uint32_t *)_sbss; dest < (uint32_t *)_ebss; )
```

Comment:
```suggestion
  for (uint32_t *dest = (uint32_t *)_sbss; dest < (uint32_t *)_ebss; )
```
Part of this change seems wrong, `dest` should still be declared as a pointer to `uint32_t`.

Reply:
Fixed.